src/main/java/fi/fmi/avi/archiver/spring/healthcontributor/AmqpConnectionHealthContributor.java

```diff
@@ -0,0 +1,73 @@
+package fi.fmi.avi.archiver.spring.healthcontributor;
+
+import com.rabbitmq.client.amqp.Connection;
+import com.rabbitmq.client.amqp.Resource;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.util.Objects.requireNonNull;
+
+public class AmqpConnectionHealthContributor implements HealthIndicator, Connection.StateListener {
+
+    private static final String DETAIL_STATUS = "status";
+    private static final String DETAIL_LAST_STATE_CHANGE = "lastStateChange";
+    private static final String DETAIL_DURATION = "duration";
+    private static final String DETAIL_ERROR = "error";
+
+    private final Clock clock;
+    private final AtomicReference<ConnectionState> lastKnownState;
```

Comment:
Looks to me that a `volatile ConnectionState` field would do here as well. Choose which one feels simpler and/or more robust.